pkg/web/apps/apps_http_handler_test.go

```diff
@@ -378,3 +381,53 @@ func Test_httpHandler_GetActiveAppByID(t *testing.T) {
 		})
 	}
 }
+
+func TestHTTPHandler_InitClientApp(t *testing.T) {
+	deviceWithMissingDeviceID := helpers.GetMockDevice() // See that just one mock is enough
+	deviceWithMissingDeviceID.DeviceID = ""              // See here that you can change the object to match with the criteria
+
+	deviceWithMissingVersion := helpers.GetMockDevice()
+	deviceWithMissingVersion.VersionID = ""
+
+	config := config.Get()
+	APIRoutePrefix := config.APIRoutePrefix
+	type fields struct {
+		Service Service
+	}
+	tests := []struct {
+		name        string
+		fields      fields
+		device      models.Device
+		mockService ServiceMock
+		wantErr     error
+		wantCode    int
+	}{
+		{
+			name:        "A 400 Bad Request should be returned when request body is missing device ID",
+			device:      *deviceWithMissingDeviceID,
+			mockService: *mockedService,
+			wantCode:    400,
+		},
+		{
+			name:        "A 400 Bad Request should be returned when request body is missing version",
+			device:      *deviceWithMissingVersion,
+			mockService: *mockedService,
+			wantCode:    400,
+		},
+	}
+	for _, tt := range tests {
+		device, _ := json.Marshal(tt.device)
+		e := echo.New()
+		req := httptest.NewRequest(http.MethodPost, "/", strings.NewReader(string(device))) // body -> device
```

Comment:
Here a new request with the tt.device parsed for a JSON in the body is made for each test.